presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushProjectionThroughExchange.java

```diff
@@ -52,6 +81,10 @@
             return Optional.empty();
         }
 
+        if (isIdentityOrNarrowingProjection(project)) {
```

Comment:
please modify assert framework for rules to check if rule is unable to be applied two times in a row